packages/svelte/src/internal/client/reactivity/props.js

```diff
@@ -267,10 +267,17 @@ export function prop(props, key, flags, fallback) {
 	// intermediate mode — prop is written to, but the parent component had
 	// `bind:foo` which means we can just call `$$props.foo = value` directly
 	if (setter) {
-		return function (/** @type {V} */ value) {
-			if (arguments.length === 1) {
-				/** @type {Function} */ (setter)(value);
-				return value;
+		var legacy_parent = props.$$legacy;
+		return function (/** @type {any} */ assigned_value, /** @type {boolean} */ mutation) {
```

Comment:
we've got `assigned_value` here and `reassigned_value` below but they're the same thing — i propose changing it to just `new_value` everywhere

```suggestion
		return function (/** @type {any} */ assigned_value, /** @type {boolean} */ mutation) {
```

Reply:
`new_value` sounds like it's the same shape as the object it's assigning to, which isn't necessarily the case. `assigned_value` for both? Or just `value` if possible?

Reply:
yeah, works for me — updated